hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java

```diff
@@ -112,31 +103,43 @@ public final void beforeRun() throws Exception {
         innerBeforeRun();
     }
 
-    private void assertNativeMapOnPartitionThread() {
-        if (!ASSERTION_ENABLED) {
-            return;
-        }
+    protected void innerBeforeRun() throws Exception {
```

Comment:
Imho better to call super instead of beforeRun/innerBeforeRun. It is the responsibility of the caller to make sure he calls the super if that is needed. Otherwise you end up with innerBeforeRun, evenMoreInnerInnerBeforeRun, trulyInnerBeforeRun

Reply:
Only methods place in this file was changed. We can address that cleanup as part of another PR